LiteCore/tests/VectorQueryTest.cc

```diff
@@ -246,6 +246,66 @@ N_WAY_TEST_CASE_METHOD(SIFTVectorQueryTest, "Hybrid Vector Query", "[Query][.Vec
+        writeMultipleTypeDocs(t);
+        t.commit();
+    }
+    createVectorIndex();
+
+    // Pre-condition: 100 documents are selected by "META().sequence % 100 = 0"
+    string                    queryStr = "SELECT count(*) FROM "s + collectionName + " WHERE META().sequence % 100 = 0";
+    Retained<Query>           query    = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    Retained<QueryEnumerator> e        = query->createEnumerator();
+    REQUIRE(e->next());
+    auto countBySecondaryWhere = e->columns()[0]->asInt();
+    REQUIRE(countBySecondaryWhere == 100);
+
+    // Plain query fails to select docs because of the additional WHERE term.
+    // ===================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target, 10) AND META().sequence % 100 = 0"
+                 " ORDER BY distance";
+    query                                         = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    Query::Options                        options = optionsWithTargetVector(kTargetVector, kData);
+    std::vector<std::pair<string, float>> result  = collectResults(query->createEnumerator(&options));
+
+    // This is plain query because of explicit max_results.
+    // VECTOR_MATCH finds first 10 docs without regard to the secondary WHERE term, META().sequence % 100 = 0,
+    // and they are filtered out by that predicate.
+    // Note: hybrid query gives the desired result, c.f. test "Hybrid Vector Query",
+    CHECK(result.size() == 0);
+
+    // Hybrid vectorsearch consistentence: as we increase the LIMIT, the results are consistent.
+    // =====================
+
+    queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+               + " WHERE VECTOR_MATCH(vecIndex, $target) AND META().sequence % 100 = 0"
+                 " ORDER BY distance";
+    query  = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+    result = collectResults(query->createEnumerator(&options));
+
+    REQUIRE(result.size() > 10);
+
+    const char* limitStrs[] = {"1", "10"};
+    const int   limits[]    = {1, 10};
+    for ( int i = 0; i < sizeof(limits) / sizeof(int); ++i ) {
+        queryStr = "SELECT META().id, VECTOR_DISTANCE(vecIndex) AS distance FROM "s + collectionName
+                   + " WHERE VECTOR_MATCH(vecIndex, $target) AND META().sequence % 100 = 0"
+                     " ORDER BY distance LIMIT "
+                   + limitStrs[i];
+        query                                         = store->compileQuery(queryStr, QueryLanguage::kN1QL);
+        std::vector<std::pair<string, float>> result0 = collectResults(query->createEnumerator(&options));
+        REQUIRE(result0.size() == limits[i]);
+        // As we increase the limit from 1 to 10, the result is consistent.
+        CHECK(std::equal(result0.begin(), result0.begin() + limits[i], result.begin()));
+    }
```

Comment:
All this is doing is testing SQLite's `LIMIT` processing; once you've established it returns ≥ 10 results without a limit, setting the limit to 1 or 10 will obviously return only that many rows. This has nothing to do with vectorsearch.

Reply:
True. I just tried to verify it, that the LIMIT is applied at the place it stands in the SQL. In vectorsearch, I see we pass LIMIT to vector_match for pure query (as an optimization). I tried to assert that the limit in the hybrid query is only used afterwards.